generic_tracers/generic_tracer_utils.F90

```diff
@@ -3272,17 +3381,7 @@ subroutine g_tracer_vertdiff_G(g_tracer, h_old, ea, eb, dt, kg_m2_to_H, m_to_H,
     !
     !   Save the current state for calculation of the implicit vertical diffusion term
     !
-
-    if (g_tracer%diag_id_vdiffuse_impl .gt. 0) then
-      if (present(mom)) then
-        do_diagnostic = .not. mom
-      else
-        do_diagnostic = .false.
-      endif
-    else
-      do_diagnostic = .false.
-    endif
-    if (do_diagnostic) then
```

Comment:
I like the idea to get rid of `do_diagnostic` and relying solely on `diag_id*` for logic control. However, this change seems to introduce an issue later on at line 3503:
https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/e6910057ac61ce6053a7a8cee877ef6ce858e921/generic_tracers/generic_tracer_utils.F90#L3503-L3517

If someone uses the Intel compiler in debug mode, it will raise an error because do_diagnostic is not defined anywhere. To resolve this, we could either uncomment lines 3276-3285 or adjust line 3503 to use `diag_id_vdiff*` as the logical condition.

Reply:
Good point. I've modified this to also use the diag ids to control whether to do the calculation of the diffusion terms in the second block, and deleted the `do_diagnostic` variable.